site/docs/expressions/scalar_functions.md

```diff
@@ -43,10 +43,68 @@ For scalar functions (one value produced for each record), a function binding de
 
 
 
+## Argument Types & Output Type Derivation
+
+### Input Argument Types
+
+Input arguments can be declared in one of three ways: complete, wildcarded, parameterized.
+
+* Complete Type: Complete types are either simple types or compound types with all parameters fully defined. Examples include i32, fp32, VARCHAR(20), List&lt;fp32&gt;, etc.
+* Wildcard Types: Compound types can be wildcarded meaning that any valid value for the parameter can be accepted. Types are wildcarded by failing to express a particular value for the type. When serialized, this will typically mean using a null pointer or sentinel value. Examples would be  VARCHAR(*), List&lt;\*&gt;, DECIMAL(\*,10).
```

Comment:
What is the difference between a parameterized type and a wildcard type in any of your examples? Is there a use case where a parameterized type doesn't cover the behavior of wildcards?

How is `List<*>` distinct from `List<T>`, for example?

Is the wildcard terminology standard outside of the database community? I haven't encountered this language before, but maybe I just haven't done enough reading.

Reply:
I would expect for example a function like `list_length`, that operates on every cell of a list-of-any-type column to have a signature of `list_length<K>(list: List<K>) -> u32`.

Reply:
I distinguished them here to specifically state that a defined parameter name must be used in an expression to be as strict/clear as possible.

Reply:
I've updated to remove wildcards.